sos/report/plugins/microovn.py

```diff
@@ -0,0 +1,150 @@
+
+    def get_tables_from_schema(self, filename, skip=None):
+        """ Get tables from schema """
+        try:
+            fname = self.path_join(filename)
+            with open(fname, 'r', encoding='UTF-8') as file:
+                try:
+                    db_schema = json.load(file)
+                except Exception:  # pylint: disable=broad-except
+                    self._log_error(f"Cannot parse JSON file {filename}")
+                    return None
+        except IOError as ex:
+            self._log_error(
+                f"Could not open DB schema file {filename}: {ex}")
+            return None
+
+        try:
+            return [table for table in dict.keys(
+                db_schema['tables']) if table not in skip]
+        except AttributeError:
+            self._log_error(f"DB schema {filename} has no 'tables' key")
+        return None
+
+    def add_database_output(self, tables, cmds, ovn_cmd):
+        """ Collect OVN database output """
+        return [f"{ovn_cmd} list {table}" for table in tables]
+
+    def setup(self):
+        # check if env is a clustered or non-clustered one
+        for pidfile in ['ovnnb_db.pid', 'ovnsb_db.pid',
+                        'ovn-northd.pid', 'ovn-controller.pid']:
+            self.add_copy_spec([
+                self.path_join('/var/snap/microovn/common/run/ovn', pidfile),
+            ])
+
+        ovn_controller_sock_path = self._find_sock(
+            self.ovn_sock_path, self.ovn_controller_sock_regex)
+
+        northd_sock_path = self._find_sock(self.ovn_sock_path,
+                                           self.ovn_northd_sock_regex)
+
+        # ovsdb nb/sb cluster status commands
+        cs = "cluster/status"
+        cos = "connection-status"
+        ovn_nb = "OVN_Northbound"
+        ovn_sb = "OVN_Southbound"
+        cmds = ""
+        self.add_cmd_output([
+            f"microovn.ovs-appctl -t {self.ovn_nbdb_sock_path} {cs} {ovn_nb}",
+            f"microovn.ovs-appctl -t {self.ovn_sbdb_sock_path} {cs} {ovn_sb}",
+            f"microovn.ovn-appctl -t {northd_sock_path} status",
+            f"microovn.ovn-appctl -t {ovn_controller_sock_path} {cos}",
+        ], foreground=True, timeout=30
+        )
+
+        # Some user-friendly versions of DB output
+        nbctl_cmds = [
+            "microovn.ovn-nbctl --no-leader-only show",
+            "microovn.ovn-nbctl --no-leader-only get-ssl",
+            "microovn.ovn-nbctl --no-leader-only get-connection",
+        ]
+
+        sbctl_cmds = [
+            "microovn.ovn-sbctl --no-leader-only show",
+            "microovn.ovn-sbctl --no-leader-only lflow-list",
+            "microovn.ovn-sbctl --no-leader-only get-ssl",
+            "microovn.ovn-sbctl --no-leader-only get-connection",
+        ]
+
+        # Database Output
+        nb_tables = self.get_tables_from_schema(self.path_join(
+            '/snap/microovn/current/share/ovn', 'ovn-nb.ovsschema'))
+
+        cmds.extend(self.add_database_output(
```

Comment:
`cmds` is a string and you can' extend a string like that; below is the error from the CI

```
Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/sos/report/__init__.py", line 1248, in setup
    plug.setup()
  File "/tmp/cirrus-ci-build/sos/report/plugins/microovn.py", line 122, in setup
    cmds.extend(self.add_database_output(
    ^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'extend'
```

Reply:
Fixed the issue.